packages/anthropic/src/anthropic-messages-language-model.ts

```diff
@@ -535,9 +524,16 @@ export class AnthropicMessagesLanguageModel implements LanguageModelV3 {
           break;
         }
         case 'tool_use': {
+          const isJsonResponseTool =
+            usesJsonResponseTool && part.name === 'json';
+
+          if (isJsonResponseTool) {
+            isJsonToolCalled = true;
+          }
```

Comment:
rm flag, use inline check for `usesJsonResponseTool` and check for tool name instead